lib/extensions/models_extensions.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:built_collection/built_collection.dart';
```

Comment:
the purpose of this extension is to keep our models as PODOs by keeping the conversion of PageData objects (PODOs we keep in our app state) to Pages (part of the Flutter framework) 

There is a PageData subclass for each page we navigate to - we keep a list of PageData objects in our app state and have a StoreConnector above the MaterialApp so that when the list changes, the MaterialApp's navigator is updated (ie. creates with a new list of pages) 

So basically we need to convert from PageData to Page, and PageData shouldn't have a dependency on Flutter so we do the conversion with an extension. 

The reason it's done in a weird way (ie. by keeping a separate map) is I found it challenging with the combination of extension functions and built_value classes and this was the best I could come up with. There may well be a much better way.

Reply:
awesome, after i went through the pr and came back to this it started making sense. the detailed comments helped so much. thanks @nickmeinhold!!!